src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -213,59 +215,52 @@ def run_custom_checks(self):
-                self.check_metadata()
-                self.nwbfile_path = str(self.save_directory / f"{self.__class__.__name__}_{num}.nwb")
+    def test_all_conversion_checks(self, setup_interface, tmp_path):
+        interface, test_name = setup_interface
 
-                self.check_no_metadata_mutation()
+        # Create a unique test name and file path
+        nwbfile_path = str(tmp_path / f"{self.__class__.__name__}_{self.test_name}.nwb")
+        self.nwbfile_path = nwbfile_path
 
-                self.check_configure_backend_for_equivalent_nwbfiles()
+        # Now run the checks using the setup objects
+        self.check_metadata_schema_valid()
+        self.check_conversion_options_schema_valid()
+        self.check_metadata()
+        self.check_no_metadata_mutation()
+        self.check_configure_backend_for_equivalent_nwbfiles()
 
-                self.check_run_conversion_in_nwbconverter_with_backend(nwbfile_path=self.nwbfile_path, backend="hdf5")
-                self.check_run_conversion_in_nwbconverter_with_backend_configuration(
-                    nwbfile_path=self.nwbfile_path, backend="hdf5"
-                )
+        self.check_run_conversion_in_nwbconverter_with_backend(nwbfile_path=nwbfile_path, backend="hdf5")
+        self.check_run_conversion_in_nwbconverter_with_backend_configuration(nwbfile_path=nwbfile_path, backend="hdf5")
 
-                self.check_run_conversion_with_backend(nwbfile_path=self.nwbfile_path, backend="hdf5")
-                self.check_run_conversion_with_backend_configuration(nwbfile_path=self.nwbfile_path, backend="hdf5")
-
-                self.check_read_nwb(nwbfile_path=self.nwbfile_path)
+        self.check_run_conversion_with_backend(nwbfile_path=nwbfile_path, backend="hdf5")
+        self.check_run_conversion_with_backend_configuration(nwbfile_path=nwbfile_path, backend="hdf5")
 
-                # TODO: enable when all H5DataIO prewraps are gone
-                # self.nwbfile_path = str(self.save_directory / f"{self.__class__.__name__}_{num}.nwb.zarr")
-                # self.check_run_conversion(nwbfile_path=self.nwbfile_path, backend="zarr")
-                # self.check_run_conversion_custom_backend(nwbfile_path=self.nwbfile_path, backend="zarr")
-                # self.check_basic_zarr_read(nwbfile_path=self.nwbfile_path)
+        self.check_read_nwb(nwbfile_path=nwbfile_path)
 
-                # Any extra custom checks to run
-                self.run_custom_checks()
+        # Any extra custom checks to run
+        self.run_custom_checks()
 
 
 class TemporalAlignmentMixin:
     """
     Generic class for testing temporal alignment methods.
-
-    This mixin must be paired with a unittest.TestCase class.
     """
 
     data_interface_cls: Type[BaseDataInterface]
-    interface_kwargs: Union[dict, list[dict]]
+    interface_kwargs: dict
+    save_directory: Path = Path(tempfile.mkdtemp())
+    conversion_options: dict = dict()
```

Comment:
Minor: should this class set a mutable default?

Reply:
Yes, you are right!